pkg/controller/core/workload_controller.go

```diff
@@ -165,10 +165,37 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	ctx = ctrl.LoggerInto(ctx, log)
 	log.V(2).Info("Reconciling Workload")
 
-	// If a deactivated workload is re-activated, we need to reset the RequeueState.
-	if wl.Status.RequeueState != nil && ptr.Deref(wl.Spec.Active, true) && workload.IsEvictedByDeactivation(&wl) {
-		wl.Status.RequeueState = nil
-		return ctrl.Result{}, workload.ApplyAdmissionStatus(ctx, r.client, &wl, true)
+	if ptr.Deref(wl.Spec.Active, true) {
+		var updated bool
+
+		// If a deactivated workload is re-activated we need to reset the RequeueState.
+		if workload.IsEvictedByDeactivation(&wl) && wl.Status.RequeueState != nil {
+			wl.Status.RequeueState = nil
+			updated = true
+		}
+
+		// If a deactivated workload is re-activated we need to set the WorkloadRequeued condition to true if already unset.
+		if workload.IsUnsetRequeuedByDeactivation(&wl) {
+			workload.SetRequeuedCondition(&wl, kueue.WorkloadRequeuedByReactivation, "The workload is reactivated")
+			updated = true
+		}
+
+		// Wait for requeueAt and set the WorkloadRequeued condition to true.
+		if condition, isUnset := workload.IsUnsetRequeuedByPodsReadyTimeout(&wl); condition != nil && isUnset {
+			var requeueAfter time.Duration
+			if wl.Status.RequeueState != nil && wl.Status.RequeueState.RequeueAt != nil {
+				requeueAfter = time.Until(wl.Status.RequeueState.RequeueAt.Time)
```

Comment:
Compute the time using the r.clock. should be available on the main branch. It allows us to use fakeClock for testing.

Reply:
Done